syntax = "proto2";
package authenticator;

message CreateAccountWeb {
    required int64 requestId = 1;
    required string email = 2;
    required string plaintextPassword = 3;
}
message CreateAccountWebSuccess {
    required int64 requestId = 1;
    required int64 userId = 2;
}
message CreateAccountWebFailure {
    required int64 requestId = 1;
    enum reason {
        ACCOUNT_ALREADY_EXISTS = 0;
        INVALID_PASSWORD = 1;
        INTERNAL_SERVER_ERROR = 2;
    }
    required reason failreason = 2;
}
message AuthenticateWeb {
    required int64 requestId = 1;
    required string email = 2;
    required string plaintextPassword = 3;
}
message AuthenticateWebSuccess {
    required int64 requestId = 1;
    required int64 userId = 2;
    required string creationDate = 3;
}
message AuthenticateWebFailure {
    required int64 requestId = 1;
    enum reason {
        INVALID_CREDENTIALS = 0;
        INTERNAL_SERVER_ERROR = 1;
    }
    required reason failreason = 2;
}
